File: easy/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0) {
    return false;
  }
  //convert to string
  const xAsString = `${x}`;

  let leftIndex = 0;
  let rightIndex = xAsString.length - 1;

  while (leftIndex <= rightIndex) {
    if (xAsString[leftIndex] != xAsString[rightIndex]) {
      return false;
    }
    leftIndex++;
    rightIndex--;
  }
  return true;
};

File: easy/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        xAsString = str(x)
        leftIndex = 0
        rightIndex = len(xAsString) - 1
        
        while(leftIndex <= rightIndex):
            if xAsString[leftIndex] != xAsString[rightIndex]:
                return False
            leftIndex += 1
            rightIndex -= 1
            
        return True
